internal/pkg/service/stream/definition/repository/branch_repo.go

```diff
@@ -50,7 +50,7 @@ func (r *BranchRepository) ExistsOrErr(k key.BranchKey) op.WithResult[bool] {
 	return r.schema.
 		Active().ByKey(k).Exists(r.client).
 		WithEmptyResultAsError(func() error {
-			return serviceError.NewResourceNotFoundError("branch", k.String(), "project")
+			return serviceError.NewResourceNotFoundError("branch", k.BranchID.String(), "project")
 		})
```

Comment:
Simplified error messages (for end user), see tests bellow, removed `<projectID>/<branchID>/...` prefix.